edb/server/conn_pool/src/metrics.rs

```diff
@@ -0,0 +1,373 @@
+use std::collections::BTreeMap;
+use std::{cell::RefCell, rc::Rc, time::Duration};
+use strum::EnumCount;
+use strum::IntoEnumIterator;
+
+use crate::algo::PoolAlgorithmDataMetrics;
+use crate::block::Name;
+
+#[derive(
+    Clone, Copy, Debug, PartialEq, Eq, Hash, strum::EnumCount, strum::EnumIter, strum::AsRefStr,
+)]
+pub enum MetricVariant {
+    Connecting,
+    Disconnecting,
+    Idle,
+    Active,
+    Failed,
+    Closed,
+    Waiting,
+}
+
+/// Maintains a rolling average of `u32` values. Note that this struct attempts
+/// to optimize `SIZE == 1`.
+#[derive(Debug, PartialEq, Eq)]
+pub struct RollingAverageU32<const SIZE: usize> {
+    values: [u32; SIZE],
+    cumulative: u64,
+    ptr: u8,
+    /// If we've never rolled over, we cannot divide the entire array by `SIZE`
+    /// and have to use `ptr` instead.
+    rollover: bool,
+}
+
+impl<const SIZE: usize> Default for RollingAverageU32<SIZE> {
+    fn default() -> Self {
+        assert!(SIZE <= u8::MAX as _);
```

Comment:
Why not use  `<const SIZE: u8>`?

Reply:
Good question. This doesn't compile when using a `u8` [1] but there might be a workaround [2] I can trying today:

[1] https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5cac05a391babb48572e722be85588f2

[2] https://github.com/rust-lang/rust/issues/104949

Reply:
Ah ok, doesn't matter then.